### ISSUE TITLE
Bad naming for kube/bookinfo-gateway.yaml

### ISSUE THREAD

**Issue by andraxylia** [istio/istio#6687]
It deploys an Ingress, not a Gateway, and can be very misleading since there is already a routing/bookinfo-gateway.yaml. At a minimu, it should be renamed to bookinfo-ingress.yaml.

In general, unclear why we need the two directories.

**Comment by frankbu** [comment]
Do we need bookinfo/ingress-gateway.yaml anymore? I think it was only needed during the transition period where we were supporting both old and new routing APIs.  think we can delete it, because in 1.0 I don't think we will document how to use Ingress, the sample will only use Gateway.

**Comment by frankbu** [comment]
We need two directories. One for the kube deployment resources (bookinfo/kube) and one for the platform-independent Istio networking resources (bookinfo/routing).

**Comment by frankbu** [comment]
And btw, all the route-rule-*.yaml files in bookinfo/kube are now obsolete, not supported, and need to be removed for 1.0.

**Comment by andraxylia** [comment]
@frankbu see my PR, I added the gateway and the ingress to the main bookinfo.yaml file.

I find the structure of the two directories /kube and /routing a bit confusing, I thought /kube is now for v1 old stuff and /routing is for v1alpha3.

Shouldn't we remove /kube completely?

**Comment by frankbu** [comment]
I'll take the TODO of cleaning up these directories as much as possible.

**Comment by vadimeisenbergibm** [comment]
@andraxylia @frankbu please note that this was exactly the plan - leave routing yamls bookinfo/kube with the original names and then remove them after 0.8.0. As an anecdote, the ingress was also gateway in the previous istio versions. 

A question - would it be useful to use helm charts for bookinfo? Different versions of micro services are required for different documentation tasks.

**Comment by andraxylia** [comment]
I personally like to run the simplest thing for bookinfo, which is a kubectl apply, and we need to install all versions anyway.

**Comment by Dmitry1987** [comment]
I was confused by the ingress 'gateway' and resource 'gateway' :) trying to make my cluster work with Istio 1 latest release. The old way of using annotation on ingress, to make Istio routing work, will be deprecated? Is the right way to use "VirtualService + Gateway + DestinationRule" resources from now on?  I tried Istio 0.6 few months ago, and now trying to recreate my previous POC found out the same manifests do not work on 1.0, and the routing started working only when I used the new resources
